source/MaterialXView/ViewerMSL.mm

```diff
@@ -0,0 +1,838 @@
+//
```

Comment:
Just a thought. Perhaps instead of splitting the viewer up into separate implementation files, it would be good to introduce a "Render Pipeline" interface, and in there it would hold things like the shaders for each pass, the main render loop, baking etc). Then the Viewer instantiates a "pipeline" object ? 

Part of the reason is that there is a lot of other logic in the Viewer that has nothing to do with the language backends in the Viewer that will have to be duplicated in 2 places while this is cleaned up.